### PR TITLE
kokkos-promotion.txt: Add comments, step 6.2

### PR DESCRIPTION
**Pull request by ndellingwood**
Add reminder comment to use the master branches of kokkos and
kokkos-kernels during snapshot into Trilinos.
Break step 6 into parts 6.1 (master branch) and 6.2 (develop branch),
      step 6.2 added to avoid forgetting merge of master back into
      develop branch so the master_history and tags carry over.

### CODE REVIEW

**Comment by ndellingwood** [comment]
Couple extra reminder notes in the doc.

File: doc/kokkos-promotion.txt [hunk 1]
```diff
@@ -175,6 +177,8 @@ Step 5: This step can be done on any SEMS machine (e.g. kokkos-dev). Actually, t
 
   5.4. Run checkin-test to push to trilinos using the CI build modules (gcc/4.9.3)
 
+     * This step is optional, the Github auto-tester is now the favored method for testing.
```
Comment (mhoemmen) [inline doc/kokkos-promotion.txt]: More than optional -- we literally aren't allowed to do it.

Comment (ibaned) [reply doc/kokkos-promotion.txt]: Good catch, @mhoemmen , this is true. We need to remove this entirely.

**Comment by ndellingwood** [comment]
@mhoemmen @ibaned I removed the checkin-script step. Forgot about this PR, but it should be ready now.

**Comment by mhoemmen** [comment]
Thanks @ndellingwood ! :D

**Comment by ndellingwood** [comment]
@ibaned or @crtrott could one of you approve and merge? Simple text changes to the promotion doc (no code changes).

**Comment by ndellingwood** [comment]
The travis-ci error is a false error (these changes touch 0 code), so this is safe to merge. 

```
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated
```

**Pull request merged**